test/e2e/common/k8s.py

```diff
@@ -82,18 +83,57 @@ def create_custom_resource(
     return _api.create_namespaced_custom_object(
         reference.group, reference.version, reference.namespace, reference.plural, custom_resource)
 
+def patch_custom_resource(
+    reference: CustomResourceReference, custom_resource: dict):
+    _api_client = _get_k8s_api_client()
+    _api = client.CustomObjectsApi(_api_client)
+
+    if reference.namespace is None:
+        return _api.patch_cluster_custom_object(
+            reference.group, reference.version, reference.plural, reference.name, custom_resource)
+    return _api.patch_namespaced_custom_object(
+        reference.group, reference.version, reference.namespace, reference.plural, reference.name, custom_resource)
+
+def delete_custom_resource(
+    reference: CustomResourceReference, wait_periods: int = 1, period_length: int = 5):
+    """Delete custom resource from cluster and wait for it to be removed by the server
+    for wait_periods * period_length seconds.
 
-def delete_custom_resource(reference: CustomResourceReference):
+    Returns:
+        response, bool:
+        response is APIserver response for the operation.
+        bool is true if resource was removed from the server and false otherwise
+    """
     _api_client = _get_k8s_api_client()
     _api = client.CustomObjectsApi(_api_client)
+
+    _response = None
     if reference.namespace is None:
-        return _api.delete_cluster_custom_object(
+        _response = _api.delete_cluster_custom_object(
             reference.group, reference.version, reference.plural, reference.name)
-    return _api.delete_namespaced_custom_object(
+    _response = _api.delete_namespaced_custom_object(
         reference.group, reference.version, reference.namespace, reference.plural, reference.name)
 
+    _removed = False
+    for _ in range(wait_periods):
+        sleep(period_length)
+        if not get_resource_exists(reference):
+            _removed = True
+            break
```

Comment:
super nit: `return _response, True` then line 127 would just be `return _response, False` and we can remove the `_removed` altogether?